### PR TITLE
Update revlib sparkutils

### LINKED ISSUE #60: Update REVLib and replace sparkutils with new features
Instead of configuration like this (from SparkModule)
```java
SparkUtils.configure(
    driveMotor,
    () ->
        SparkUtils.configureFrameStrategy(
            driveMotor,
            Set.of(Data.POSITION, Data.VELOCITY, Data.APPLIED_OUTPUT),
            Set.of(Sensor.INTEGRATED),
            false),
    () -> driveMotor.setIdleMode(IdleMode.kBrake),
    () -> driveMotor.setSmartCurrentLimit((int) Driving.CURRENT_LIMIT.in(Amps)),
    () -> driveEncoder.setPositionConversionFactor(Driving.POSITION_FACTOR.in(Meters)),
    () -> driveEncoder.setVelocityConversionFactor(Driving.VELOCITY_FACTOR.in(MetersPerSecond)),
    () -> driveEncoder.setAverageDepth(16),
    () -> driveEncoder.setMeasurementPeriod(32));
```
go back to the system of
```java
check(driveMotor, driveMotor.restoreFactoryDefaults());
check(driveMotor, driveMotor.setIdleMode(IdleMode.kBrake));
...
check(driveMotor, SparkUtils.configureFrameStrategy(...));
check(driveMotor, driveMotor.burnFlash());
```
where check is `FaultLogger.check`
Additionally, we should be able to remove `SparkUtils.configure` `SparkUtils.wrapCall` and `SparkUtils.setInverted`

### PR DESCRIPTION
**Pull request by HenryLi-0**
for Issue #60

### CODE REVIEW

File: src/main/java/org/sciborgs1155/lib/FaultLogger.java [hunk 1]
```diff
@@ -274,6 +274,14 @@ public static boolean check(CANSparkBase spark) {
     return true;
   }
 
+  public static boolean check(CANSparkBase spark, REVLibError error) {
```
Comment (anglesideangle) [inline src/main/java/org/sciborgs1155/lib/FaultLogger.java:282]: Add javadoc to this function (you can copy paste from the other `check` and add another `@param`). Also, make the other `check(CANSparkBase)` call this method to use less code

File: src/main/java/org/sciborgs1155/robot/drive/SparkModule.java [hunk 2]
```diff
@@ -41,40 +41,48 @@ public SparkModule(int drivePort, int turnPort, Rotation2d angularOffset) {
     driveMotor = new CANSparkFlex(drivePort, MotorType.kBrushless);
     driveEncoder = driveMotor.getEncoder();
 
-    SparkUtils.configure(
+    FaultLogger.check(driveMotor, driveMotor.restoreFactoryDefaults());
```
Comment (anglesideangle) [inline src/main/java/org/sciborgs1155/robot/drive/SparkModule.java]: this applies everywhere, you can statically import `FaultLogger.check`, and then just write `check(...)` to make things more readable

File: src/main/java/org/sciborgs1155/robot/drive/SparkModule.java [hunk 3]
```diff
@@ -41,40 +41,48 @@ public SparkModule(int drivePort, int turnPort, Rotation2d angularOffset) {
...
+            driveMotor,
+            Set.of(Data.POSITION, Data.VELOCITY, Data.APPLIED_OUTPUT),
+            Set.of(Sensor.INTEGRATED),
+            false));
+    FaultLogger.check(driveMotor, driveMotor.burnFlash());
 
     turnMotor = new CANSparkMax(turnPort, MotorType.kBrushless);
     turningEncoder = turnMotor.getAbsoluteEncoder();
 
-    SparkUtils.configure(
```
Comment (anglesideangle) [inline src/main/java/org/sciborgs1155/robot/drive/SparkModule.java:62]: make sure to restore factory defaults and burn flash every spark

File: src/main/java/org/sciborgs1155/robot/intake/RealIntake.java [hunk 4]
```diff
@@ -19,12 +19,11 @@ public class RealIntake implements IntakeIO {
...
 
   public RealIntake() {
-    SparkUtils.configure(
-        spark,
-        () -> SparkUtils.configureNothingFrameStrategy(spark),
-        () -> SparkUtils.setInverted(spark, true),
-        () -> spark.setIdleMode(IdleMode.kBrake),
-        () -> spark.setSmartCurrentLimit((int) CURRENT_LIMIT.in(Amps)));
+    FaultLogger.check(spark, SparkUtils.configureNothingFrameStrategy(spark));
+    spark.setInverted(true);
```
Comment (anglesideangle) [inline src/main/java/org/sciborgs1155/robot/intake/RealIntake.java:24]: check this using `check(spark)`

**Review by anglesideangle — CHANGES_REQUESTED** [review]

File: src/main/java/org/sciborgs1155/robot/intake/RealIntake.java (hunk 4 above)
Comment (HenryLi-0) [reply src/main/java/org/sciborgs1155/robot/intake/RealIntake.java:24]: `spark.setInverted(true);` returns void

**Pull request merged**